bpmn-model/src/main/java/io/camunda/zeebe/model/bpmn/validation/zeebe/MessageValidator.java

```diff
@@ -112,6 +117,26 @@ private boolean isReferredByCatchEvent(final Message element) {
                     && ((MessageEventDefinition) e).getMessage() == element);
   }
 
+  private boolean isReferredByThrowEvent(final Message element) {
+    final Collection<IntermediateThrowEvent> intermediateCatchEvents =
+        getAllElementsByType(element, IntermediateThrowEvent.class);
+
+    final Collection<EndEvent> endEvents = getAllElementsByType(element, EndEvent.class);
+
+    return Stream.concat(intermediateCatchEvents.stream(), endEvents.stream())
+        .flatMap(i -> i.getEventDefinitions().stream())
+        .anyMatch(
+            e ->
+                e instanceof MessageEventDefinition
+                    && ((MessageEventDefinition) e).getMessage() == element);
```

Comment:
💭 Not sure if this is actually better, but here's an alternative:

```java
.filter(MessageEventDefinition.class::isInstance)
.map(MessageEventDefinition.class::cast)
.anyMatch(e -> e.getMessage() == element);
```

And another 🤔 Maybe I like this one better

```java
.filter(MessageEventDefinition.class::isInstance)
.anyMatch(e -> ((MessageEventDefinition) e).getMessage() == element);
```

Feel free to ignore this